flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/plan/schema/IntermediateRelTable.scala

```diff
@@ -54,4 +55,19 @@ class IntermediateRelTable(
   def this(names: JList[String], relNode: RelNode) {
     this(names, relNode, ModifyKindSet.INSERT_ONLY, false, new util.HashSet[ImmutableBitSet]())
   }
+
+  override def getCollationList: util.List[RelCollation] = {
+    val mq = relNode.getCluster.getMetadataQuery
+    mq.collations(relNode)
+  }
+
+  override def getDistribution: RelDistribution = {
+    val fmq = FlinkRelMetadataQuery.reuseOrCreate(relNode.getCluster.getMetadataQuery)
```

Comment:
> Why don't use
> 
> ```
>     val fmq = relNode.getCluster.getMetadataQuery
>     fmq.distribution(relNode)
> ```

For `RelDistribution`, we must convert `RelMetadataQuery` to `FlinkRelMetadataQuery`, and call `fmq.flinkDistribution(relNode)` instead of `fmq.distribution(relNode)` to get `RelDistribution` (Here is a bit confusing. I still need to have a good understand).

Reply:
> Why don't use
> 
> ```
>     val fmq = relNode.getCluster.getMetadataQuery
>     fmq.distribution(relNode)
> ```

The implementation of FlinkDistribution and Calcite Distribution is different, and theoretically they need to be the same. I can create another PR to unify them.